python/meep.i

```diff
@@ -765,6 +776,7 @@ void _get_gradient(PyObject *grad, PyObject *fields_a, PyObject *fields_f, PyObj
     if (!PyArray_Check(pao_grad)) meep::abort("grad parameter must be numpy array.");
     if (!PyArray_ISCARRAY(pao_grad)) meep::abort("Numpy grad array must be C-style contiguous.");
     meep::realnum * grad_c = (meep::realnum *)PyArray_DATA(pao_grad);
+    int ng = PyArray_DIMS(pao_grad)[1];
```

Comment:
do we check that `PyArray_NDIM(pao_grad) == 1`?

Reply:
We dont but probably should

Reply:
Should be fixed now